Directory.cpp

```diff
@@ -6,13 +6,21 @@ this->name=n;
 this->parent=p;
 }
 Directory::~Directory(){
-for(auto& i:subFolders){
-    delete i;
-}
+    for(auto& i:subFolders){
+        delete i;
+    }
+    for(auto& i: files){
+        delete i;
+    }
 }
 string Directory::getName() const{
 return name;
 }
+void Directory::echo(string content, string filename){
+    File * file = searchFile(filename);
+    file->setContent(content);
+    return;
```

Comment:
Azért is kéne idetenni az ellenőrzést, mert ez OO szempontból is hibás. Nem várhatjuk el a paraméterről, hogy az létező fájlnevet tartalmaz.

+1 felesleges return